File: JavaScript/isPalindrome.js
const isPalindrome = s => {

    // sanitize input string - lowercase and regex \W character class
    s = s.toLowerCase().replace(/[\W]/g, "");
    
    let start = 0;
    let end = s.length-1;

    for(let i = 0; i < s.length; i++) {
        if(s[start] !== s[end]) {
            return false;
        }

        start++;
        end--;
    }

    return true;
}

console.log(isPalindrome("A man, a plan, a canal: Panama"));

File: Java/IsPalindrome.java
public class IsPalindrome {
    public static boolean isPalindrome(String s) {

        s = s.toLowerCase().replaceAll("\\W", "");
        System.out.println(s);
        int start = 0;
        int end = s.length()-1;

        for(int i = 0; i < s.length(); i++) {
            if(s.charAt(start) != s.charAt(end)) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        System.out.println(isPalindrome("A man, a plan, a canal: Panama"));
    }
}
